File: LeetCode/Is Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        index = 0
        
        for i in range(len(t)):
            if index < len(s) and s[index] == t[i]:
                index += 1
        
        return index == len(s)
        
        

File: LeetCode/Is Subsequence.go
func isSubsequence(s string, t string) bool {
    index := 0
    
    for i:=0; i < len(t); i++ {
        if (index < len(s) && s[index] == t[i]) {
            index++
        }
    }
    
    return index == len(s)
}